libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -320,6 +320,58 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
             await connectorClient.Conversations.DeleteActivityAsync(reference.Conversation.Id, reference.ActivityId);
         }
 
+        /// <summary>
+        /// Attempts to retrieve the token for a user that's in a login flow.
+        /// </summary>
+        /// <param name="context">Context for the current turn of conversation with the user.</param>
+        /// <param name="connectionName">Name of the auth connection to use.</param>
+        /// <param name="magicCode">(Optional) Optional user entered code to validate.</param>
+        /// <returns></returns>
+        public async Task<TokenResponse> GetUserToken(ITurnContext context, string connectionName, string magicCode)
```

Comment:
Consider refactoring these methods to an interface, so they could be added to other Adapters. I can easily see the Alexa Adapter needthing this behavior. Even the ConsoleAdapter may want this.

Reply:
good idea! but will leave that for next iteration.

---
In reply to: [184821299](https://github.com/Microsoft/botbuilder-dotnet/pull/520#discussion_r184821299) [](ancestors = 184821299)